src/features/fee-row/fee-row.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { Suspense, useCallback, useEffect, useMemo, useState } from 'react';
+import { FiInfo } from 'react-icons/fi';
+import { useField } from 'formik';
+import { Box, color, Flex, Stack, Tooltip } from '@stacks/ui';
+
+import { stacksValue } from '@common/stacks-utils';
+import { openInNewTab } from '@common/utils/open-in-new-tab';
+import { LoadingRectangle } from '@components/loading-rectangle';
+import { SpaceBetween } from '@components/space-between';
+import { Caption } from '@components/typography';
+import { Estimations } from '@models/fees-types';
+import {
+  useFeeEstimationsState,
+  useFeeRateState,
+  useFeeState,
+} from '@store/transactions/fees.hooks';
+
+import { TransactionFee } from './components/transaction-fee';
+import { FeeEstimateItem } from './components/fee-estimate-item';
+import { FeeEstimateSelect } from './components/fee-estimate-select';
+import { CustomFeeField } from './components/custom-fee-field';
+import { SendFormErrorMessages } from '@common/error-messages';
+import { CustomFeeError } from './components/custom-fee-error';
+
+interface FeeRowProps {
+  isError: boolean;
+}
+
+export function FeeRow(props: FeeRowProps): JSX.Element {
+  const { isError } = props;
+  const [, meta, helpers] = useField('txFee');
+  const [isOpen, setIsOpen] = useState(false);
+  const [feeEstimations] = useFeeEstimationsState();
+  const [fee, setFee] = useFeeState();
+  const [, setFeeRate] = useFeeRateState();
+  const [selected, setSelected] = useState(Estimations.Middle);
+  const [isCustom, setIsCustom] = useState(false);
+  const feesInfo =
+    'Higher fees increase the likelihood of your transaction getting confirmed before others. Click to learn more.';
+  const url = 'https://hiro.so/questions/fee-estimates';
```

Comment:
I would tend to move these out of the component (or write inline), as they're redeclared on every render. Not a biggie, but a small optimisation.

Reply:
Yep, good point, thx.